Fix all pep8 and pyflakes failures, add helper run_tests script

ahayzen opened the pull request:
Understand if you don't want this, not everyone wants 79 char line lengths :-)

Before
```
$ pep8 .
./src/checker.py:27:1: E302 expected 2 blank lines, found 1
./src/checker.py:30:1: E302 expected 2 blank lines, found 1
./src/checker.py:37:80: E501 line too long (81 > 79 characters)
./src/checker.py:38:80: E501 line too long (82 > 79 characters)
./src/checker.py:41:80: E501 line too long (84 > 79 characters)
./src/checker.py:42:80: E501 line too long (81 > 79 characters)
./src/checker.py:43:80: E501 line too long (87 > 79 characters)
./src/checker.py:48:80: E501 line too long (82 > 79 characters)
./src/checker.py:49:80: E501 line too long (89 > 79 characters)
./src/checker.py:54:75: E502 the backslash is redundant between brackets
./src/checker.py:99:18: E201 whitespace after '{'
./src/checker.py:122:25: E703 statement ends with a semicolon
./src/lib/externaldata.py:27:1: E302 expected 2 blank lines, found 1
./src/lib/externaldata.py:40:23: E261 at least two spaces before inline comment
./src/lib/externaldata.py:41:24: E261 at least two spaces before inline comment
./src/lib/externaldata.py:70:80: E501 line too long (81 > 79 characters)
./src/lib/externaldata.py:91:1: E302 expected 2 blank lines, found 1
./src/lib/externaldata.py:96:1: E302 expected 2 blank lines, found 1
./src/lib/externaldata.py:102:80: E501 line too long (82 > 79 characters)
./src/lib/utils.py:23:1: E302 expected 2 blank lines, found 1
./src/lib/utils.py:31:1: E302 expected 2 blank lines, found 1
./src/lib/utils.py:36:1: E302 expected 2 blank lines, found 1
./src/checkers/debianrepochecker.py:45:1: E302 expected 2 blank lines, found 1
./src/checkers/debianrepochecker.py:74:80: E501 line too long (85 > 79 characters)
./src/checkers/debianrepochecker.py:76:1: E302 expected 2 blank lines, found 1
./src/checkers/debianrepochecker.py:88:80: E501 line too long (90 > 79 characters)
./src/checkers/debianrepochecker.py:158:51: E128 continuation line under-indented for visual indent
./src/checkers/debianrepochecker.py:179:1: E305 expected 2 blank lines after class or function definition, found 1
./src/checkers/rotatingurlchecker.py:37:1: E302 expected 2 blank lines, found 1
./src/checkers/rotatingurlchecker.py:46:80: E501 line too long (91 > 79 characters)
./src/checkers/rotatingurlchecker.py:69:1: E305 expected 2 blank lines after class or function definition, found 1
./src/checkers/urlchecker.py:27:1: E302 expected 2 blank lines, found 1
./src/checkers/urlchecker.py:32:9: E722 do not use bare except'
./src/checkers/urlchecker.py:37:1: E305 expected 2 blank lines after class or function definition, found 1
./tests/test_checker.py:29:1: E402 module level import not at top of file
./tests/test_checker.py:30:1: E402 module level import not at top of file
./tests/test_checker.py:32:80: E501 line too long (80 > 79 characters)
./tests/test_checker.py:39:1: E302 expected 2 blank lines, found 1
./tests/test_checker.py:42:80: E501 line too long (87 > 79 characters)
./tests/test_checker.py:45:1: E302 expected 2 blank lines, found 1
./tests/test_checker.py:87:80: E501 line too long (84 > 79 characters)
./tests/test_checker.py:90:80: E501 line too long (81 > 79 characters)
./tests/test_checker.py:96:1: E305 expected 2 blank lines after class or function definition, found 1

$ pyflakes3 .
./src/lib/externaldata.py:75: undefined name '__class__'
./src/lib/externaldata.py:76: undefined name '__class__'
```

After

```
$ pep8 .
$ pyflakes3 .
```

I also added a `run_tests.sh` script to set the PYTHONPATH instead of using the `sys` module, maybe there should be a `run_linters.sh` or something if you accept this for pep8 + pyflakes3 ?

ahayzen committed:
Fix all pep8 and pyflakes failures, add helper run_tests script

ahayzen committed:
Fix all pep8 and pyflakes failures, add helper run_tests script

ahayzen committed:
Add gitignore for __pycache__ folder (#4)

joaquimrocha commented on src/checker.py:
```diff
@@ -24,34 +24,43 @@
     def _collect_external_data(self):
-        self._external_data = self._get_module_data_from_json(self._json_data) + \
-                              self._get_finish_args_extra_data_from_json(self._json_data)
+        self._external_data = (
```
Is this expected by pep8? I really don't love this change.

joaquimrocha commented on src/checker.py:
```diff
@@ -119,7 +128,7 @@ def check(self, filter_type=None):
         for data in self._external_data:
             # Ignore if the type is not the one we care about
             if filter_type is not None and filter_type != data.type:
-                continue;
```
Oops.

joaquimrocha commented on src/lib/externaldata.py:
```diff
@@ -53,27 +54,28 @@ def __init__(self, data_type, filename, url, checksum, size=-1, arches=[],
-                                                  size=self.size,
-                                                  arches=self.arches,
-                                                  checker_data=self.checker_data)
+        info = (
```
Again, I don't love this change.

joaquimrocha commented on src/lib/externaldata.py:
```diff
@@ -53,27 +54,28 @@ def __init__(self, data_type, filename, url, checksum, size=-1, arches=[],
         json_data = OrderedDict()
-        json_data['type'] = __class__._TYPES_MANIFEST_MAP[self.type]
-        json_data[__class__._TYPES_MANIFEST_MAP[self.type]] = self.filename
+        json_data['type'] = self._TYPES_MANIFEST_MAP[self.type]
```
Did you run this in Python 3? I just wanted to use the class itself here.

joaquimrocha commented on tests/test_checker.py:
```diff
@@ -84,14 +83,17 @@ def test_check(self):
 
-        archive_ext_data = self.checker.get_external_data(ExternalData.Type.ARCHIVE)
+        archive_ext_data = self.checker.get_external_data(
+            ExternalData.Type.ARCHIVE)
```
I don't like that this is to the left of the `=` sign. Does it fit if you bring the `self.checker...` to the line below?

joaquimrocha reviewed (changes requested):
I like the general idea of having a lint in order to make the code coherent. For Python I think usually the code already turns out very consistent anyway, but I don't oppose to using a lint here either. It does seem a bit too harsh to enforce 80 chars in today's displays, but it's the weird formatting to accommodate the limit that I dislike. Please check my comments, and I can accept this if in the end the formatting is not completely crazy.

joaquimrocha commented on src/lib/externaldata.py:
```diff
@@ -88,18 +90,21 @@ def to_json(self):
     def load(checkers_folder):
-        for _unused, modname, _unused in pkgutil.walk_packages([checkers_folder]):
+        for _unused, modname, _unused in pkgutil.walk_packages(
+                [checkers_folder]):
```
I don't like opening parenthesis without args in that line. I'd prefer to break after the `in`.

ramcq committed:
DebianRepoChecker: add support for Packages.gz2

Also, prefer .xz over .bz2 over uncompressed.

ramcq committed:
DebianRepoChecker: add support for exact repo URLs

Debian upstream repos have the dists/{dist}/../{component} structure, but apt
sources.list also supports a syntax with a base URL followed by a subdirectory:

  "suite can specify an exact path, in which case the components must be
  omitted and suite must end with a slash (/). This is useful for the case when
  only a particular sub-directory of the archive denoted by the URI is of
  interest. If suite does not specify an exact path, at least one component
  must be present."

Add support for omitting the "component" if "dist" ends in / to allow checking
repos with this structure.

ramcq committed:
DebianRepoChecker: print a warning if component is missing

Now that component is optional, if dist does not end in a / specifying
an exact path, then the configuration is invalid. Detect this case and
print a warning.

ahayzen committed:
Change max line length to 100 and add run_lints.sh script

ahayzen committed:
Change max line length to 100 and add run_lints.sh script

ahayzen committed:
Extra fix after rebase

ahayzen committed:
Merge branch 'pep8-pyflakes' of github.com:ahayzen/flatpak-external-data-checker into pep8-pyflakes

ahayzen committed:
Further fix

ahayzen committed:
Another revert due to longer lines
